packages/server/src/api/ampEpicRouter.ts

```diff
@@ -0,0 +1,236 @@
+                                  MONTHLY: ' per month',
+                                  ANNUAL: ' per year',
+                              },
+                              classNames: {
+                                  choiceCard: 'epicChoiceCard',
+                                  choiceCardSelected: 'epicChoiceCard epicChoiceCardSelected',
+                              },
+                              currencySymbol: getLocalCurrencySymbol(countryCode),
+                          }
+                        : false,
+                };
+
+                res.setHeader('Content-Type', 'application/json');
+                res.setHeader('Cache-Control', 'private, no-store');
+                res.setHeader('Surrogate-Control', 'max-age=0');
+
+                res.json({ ...epic, ...ampState });
+            } catch (error) {
+                next(error);
+            }
+        },
+    );
+
+    router.get(
+        '/amp/epic_view', // IMPORTANT: do not change this route!
+        cors({
+            origin: [
+                'https://amp-theguardian-com.cdn.ampproject.org',
+                'https://amp.theguardian.com',
+                'http://localhost:3030',
+                'https://amp.code.dev-theguardian.com',
+            ],
+            credentials: true,
+            allowedHeaders: ['x-gu-geoip-country-code'],
+        }),
+        async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            try {
+                res.setHeader('Cache-Control', 'private, no-store');
+                res.setHeader('Surrogate-Control', 'max-age=0');
+
+                const countryCode = req.header('X-GU-GeoIP-Country-Code');
+                const ampVariantAssignments = getAmpVariantAssignments(req);
+                const tests = await getCachedAmpEpicTests();
+                const epic = await ampEpic(tests, ampVariantAssignments, countryCode);
+                const campaignCode = buildAmpEpicCampaignCode(epic.testName, epic.variantName);
+                const { viewId, ampViewId, browserIdCookie, browserId } = req.query;
+
+                const browserIdQuery =
+                    browserIdCookie && browserId ? `&${browserIdCookie}=${browserId}` : '';
+
+                const ophanComponentEvent: OphanComponentEvent = {
+                    component: {
+                        componentType: 'ACQUISITIONS_EPIC',
+                        products: ['CONTRIBUTION', 'MEMBERSHIP_SUPPORTER'],
+                        campaignCode: campaignCode,
+                        id: campaignCode,
+                    },
+                    abTest: {
+                        name: epic.testName,
+                        variant: epic.variantName,
+                    },
+                    action: 'VIEW',
+                };
+
+                const ophanUrl = `https://ophan.theguardian.com/img/2?viewId=${viewId}&ampViewId=${ampViewId}${browserIdQuery}&componentEvent=${encodeURI(
```

Comment:
Oh, interesting that this happens server side and not on the client! Is that because this is amp and we're more restricted in what we can do on the client? (Sorry, tangential question as this hasn't changed on this PR!)

Reply:
yes, the AMP epic is a big hack! Here it's adding the `VIEW` componentEvent to the existing pageview record